Jenkinsfile

```diff
@@ -16,55 +16,126 @@ pipeline {
-        stage('Publish pre-release images from pull request') {
-            when {
-                changeRequest target: 'master'
+        stage('Publish in fromdoppler') {
+            environment {
+                DOCKER_CREDENTIALS_ID = "dockerhub_fromdoppler"
+                DOCKER_IMAGE_NAME = "fromdoppler/doppler-for-shopify"
             }
-            steps {
-                sh 'sh build-n-publish.sh --commit=${GIT_COMMIT} --name=pr-${CHANGE_ID}'
-            }
-        }
-        stage('Publish pre-release images from master') {
-            when {
-                branch 'master'
-            }
-            steps {
-                sh 'sh build-n-publish.sh --commit=${GIT_COMMIT} --name=master'
-            }
-        }
-        stage('Publish pre-release images from INT') {
-            when {
-                branch 'INT'
-            }
-            steps {
-                sh 'sh build-n-publish.sh --commit=${GIT_COMMIT} --name=INT'
-            }
-        }
-        stage('Publish pre-release images from PROD') {
-			// Temporal build for testing easily in production
-            when {
-                branch 'PROD'
-            }
-            steps {
-                sh 'sh build-n-publish.sh --commit=${GIT_COMMIT} --name=PROD'
-            }
-        }
-        stage('Publish final version images') {
-            when {
-                expression {
-                    return isVersionTag(readCurrentTag())
+            stages {
+                stage('Publish pre-release images from pull request') {
+                    when {
+                        changeRequest target: 'master'
+                    }
+                    steps {
+                        withDockerRegistry(credentialsId: "${DOCKER_CREDENTIALS_ID}", url: "") {
+                            sh 'sh build-n-publish.sh --image=${DOCKER_IMAGE_NAME} --commit=${GIT_COMMIT} --name=pr-${CHANGE_ID}'
+                        }
+                    }
+                }
+                stage('Publish pre-release images from master') {
+                    when {
+                        branch 'master'
+                    }
+                    steps {
+                        withDockerRegistry(credentialsId: "${DOCKER_CREDENTIALS_ID}", url: "") {
+                            sh 'sh build-n-publish.sh --image=${DOCKER_IMAGE_NAME} --commit=${GIT_COMMIT} --name=master'
+                        }
+                    }
+                }
+                stage('Publish pre-release images from INT') {
+                    when {
+                        branch 'INT'
+                    }
+                    steps {
+                        withDockerRegistry(credentialsId: "${DOCKER_CREDENTIALS_ID}", url: "") {
+                            sh 'sh build-n-publish.sh --image=${DOCKER_IMAGE_NAME} --commit=${GIT_COMMIT} --name=INT'
+                        }
+                    }
+                }
+                stage('Publish pre-release images from PROD') {
```

Comment:
I known that this is not relevant in this PR. Is this stage still necessary ?

Reply:
Mmm, I do not remember. It was useful for me last time that I had to touch this repository, so, I think that we can keep it for the next time.